### PR TITLE
[Team Email Login] Genericize AdminConfig Modal

### PR DESCRIPTION
**Pull request by msarit**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Re: @vaidehijoshi's comment on a [previous PR](https://github.com/forem/forem/pull/11090), this PR genericizes the `adminConfigModal` so that it can be used throughout the Admin console.

## Related Tickets & Documents
- https://github.com/forem/InternalProjectPlanning/issues/244

## QA Instructions, Screenshots, Recordings

None really needed

## Added tests?

- [ ] Yes
- [X] No, and this is why: _PR simply abstracts a function (which builds the modal)_
- [ ] I need help with writing tests

## Added to documentation?

- [ ] Docs.forem.com
- [ ] README
- [X] No documentation needed

## [optional] Are there any post deployment tasks we need to perform?
No

### CODE REVIEW

**Review by joshpuetz — APPROVED** [review]
👍

File: app/javascript/admin/genericAdminModal.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const genericAdminModal = (
```
Comment (vaidehijoshi) [inline app/javascript/admin/genericAdminModal.js:1]:
```suggestion
const adminModal = (
```
I think `generic` is a bit redundant in the naming here. Thinking about how we name other reusable components across our codebase (like crayons), and we opt to just call them `Modal` for example.

Comment (vaidehijoshi) [inline app/javascript/admin/genericAdminModal.js:1]: Can we add some documentation for how this modal works and how it should be used?

We've been using [JSDoc](https://jsdoc.app/about-getting-started.html) for JS documentation; the [initializeBroadcast.js](https://github.com/forem/forem/blob/master/app/assets/javascripts/initializers/initializeBroadcast.js) file has some examples of how we've been documenting in other JS files if you want a good starting point :)

**Comment by vaidehijoshi** [comment]
@msarit could we add some tests around this modal? I know that you added tests around the site config work in https://github.com/forem/forem/pull/11090, but since we're starting to abstract the code and make it reusable, i think that it would also be a good time to add modal-specific tests so that we can catch any changes to this modal since we're allowing it to be reused in other places :)

File: app/javascript/admin/genericAdminModal.js (hunk 1 above)
Comment (joshpuetz) [reply app/javascript/admin/genericAdminModal.js]: Oh WOW, thanks for pointing this out: I didn't know we were using something for JS documentation!

**Comment by msarit** [comment]
Flaky, unrelated specs are plaguing this PR 😩 
`spec/system/comments/user_edits_a_comment_spec.rb`
`spec/system/articles/user_creates_an_article_spec.rb`

**Review by nickytonline — APPROVED** [review]
LGTM @msarit! 🚀

**Pull request merged**